sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StreamingWriteTables.java

```diff
@@ -105,8 +105,8 @@ public WriteResult expand(PCollection<KV<TableDestination, TableRow>> input) {
                 "StreamingWrite",
                 ParDo.of(new StreamingWriteFn(bigQueryServices, retryPolicy, failedInsertsTag))
                     .withOutputTags(mainOutputTag, TupleTagList.of(failedInsertsTag)));
-    PCollection<TableRow> failedInserts = tuple.get(failedInsertsTag);
-    failedInserts.setCoder(TableRowJsonCoder.of());
+    PCollection<BigQueryInsertError> failedInserts = tuple.get(failedInsertsTag);
+    failedInserts.setCoder(BigQueryInsertErrorCoder.of());
```

Comment:
This is an update-incompatible change of the type we try to avoid. Many users of streaming rely on the ability to do online updates of running pipelines. Changing the type of a PCollection breaks that (the update will be rejected as incompatible), so we try to avoid this. The best solution might be to simply create a new PCollection and leave the old one around (but not published to).

Reply:
👍